### PR TITLE
Handle user decryption errors in DecryptionQueue

### PR DESCRIPTION
**Pull request by bolekk**

### CODE REVIEW

File: core/services/ocr2/plugins/threshold/decryption_queue.go [hunk 1]
```diff
@@ -206,20 +211,30 @@ func (dq *decryptionQueue) SetResult(ciphertextId decryptionPlugin.CiphertextId,
 
 	req, ok := dq.pendingRequests[string(ciphertextId)]
 	if ok {
-		dq.lggr.Debugf("responding with result for pending decryption request ciphertextId %s", string(ciphertextId))
-		req.chPlaintext <- plaintext
+		if err != nil {
+			dq.lggr.Errorw("decryption error", "ciphertextId", toHex(ciphertextId), "error", err)
```
Comment (KuphJr) [inline core/services/ocr2/plugins/threshold/decryption_queue.go:215]: Does this mean we will log errors if a user incorrectly encrypts their secrets?

Comment (bolekk) [reply core/services/ocr2/plugins/threshold/decryption_queue.go:215]: Yes. Are you suggesting making it a debug message? You might be right... We're not very consistent with that, for example we log ERROR for invalid CBOR or request-too-big.

I'd agree that it's a better idea to simply log debug messages for all user errors. But maybe let's try to add some extra prom counters so we can notice and investigate those.

File: core/services/ocr2/plugins/threshold/decryption_queue.go [hunk 2]
```diff
@@ -61,6 +62,10 @@ func NewDecryptionQueue(maxQueueLength int, maxCiphertextBytes int, maxCiphertex
 	return &dq
 }
 
+func toHex(ciphertextId []byte) string {
+	return "0x" + hex.EncodeToString(ciphertextId)
```
Comment (jmank88) [inline core/services/ocr2/plugins/threshold/decryption_queue.go:66]: Would it be simpler to add this as a `String()` method for `decryptionPlugin.CiphertextId`?

Comment (bolekk) [reply core/services/ocr2/plugins/threshold/decryption_queue.go:66]: Good suggestion, thanks :)

**Comment by bolekk** [comment]
I'll address logging problems separately (FUN-814) to keep this PR concise.

File: core/services/ocr2/plugins/threshold/decryption_queue.go (hunk 2 above)
Comment (bolekk) [reply core/services/ocr2/plugins/threshold/decryption_queue.go]: Actually I'll address that in a separate PR.

File: core/services/ocr2/plugins/threshold/decryption_queue.go [hunk 3]
```diff
@@ -206,11 +206,21 @@ func (dq *decryptionQueue) SetResult(ciphertextId decryptionPlugin.CiphertextId,
 
 	req, ok := dq.pendingRequests[string(ciphertextId)]
 	if ok {
-		dq.lggr.Debugf("responding with result for pending decryption request ciphertextId %s", string(ciphertextId))
-		req.chPlaintext <- plaintext
+		if err != nil {
+			dq.lggr.Debugf("decryption error for ciphertextId %s", string(ciphertextId))
+		} else {
+			dq.lggr.Debugf("responding with result for pending decryption request ciphertextId %s", string(ciphertextId))
+			req.chPlaintext <- plaintext
```
Comment (stchrysa) [inline core/services/ocr2/plugins/threshold/decryption_queue.go:213]: Check maybe if plaintext is nil?

File: core/services/ocr2/plugins/threshold/decryption_queue.go [hunk 4]
```diff
@@ -206,11 +206,21 @@ func (dq *decryptionQueue) SetResult(ciphertextId decryptionPlugin.CiphertextId,
 
 	req, ok := dq.pendingRequests[string(ciphertextId)]
 	if ok {
-		dq.lggr.Debugf("responding with result for pending decryption request ciphertextId %s", string(ciphertextId))
-		req.chPlaintext <- plaintext
+		if err != nil {
+			dq.lggr.Debugf("decryption error for ciphertextId %s", string(ciphertextId))
```
Comment (stchrysa) [inline core/services/ocr2/plugins/threshold/decryption_queue.go:210]: Is there a reason for suppressing the error message?

File: core/services/ocr2/plugins/threshold/decryption_queue.go [hunk 5]
```diff
@@ -206,11 +206,21 @@ func (dq *decryptionQueue) SetResult(ciphertextId decryptionPlugin.CiphertextId,
...
+		} else {
+			dq.lggr.Debugf("responding with result for pending decryption request ciphertextId %s", string(ciphertextId))
+			req.chPlaintext <- plaintext
+		}
 		close(req.chPlaintext)
 		delete(dq.pendingRequests, string(ciphertextId))
 	} else {
+		if err != nil {
+			// This is currently possible only for ErrAggregation, encountered during Report() phase.
+			dq.lggr.Debugf("received decryption error for ciphertextId %s which doesn't exist locally", string(ciphertextId))
```
Comment (stchrysa) [inline core/services/ocr2/plugins/threshold/decryption_queue.go:220]: Same as above, why suppress the error message?

File: core/services/ocr2/plugins/threshold/decryption_queue.go (hunk 3 above)
Comment (bolekk) [reply core/services/ocr2/plugins/threshold/decryption_queue.go:213]: Will do.

File: core/services/ocr2/plugins/threshold/decryption_queue.go (hunk 5 above)
Comment (bolekk) [reply core/services/ocr2/plugins/threshold/decryption_queue.go:220]: See above.

**Review by stchrysa — APPROVED** [review]

File: core/services/ocr2/plugins/threshold/decryption_queue.go (hunk 1 above)
Comment (KuphJr) [reply core/services/ocr2/plugins/threshold/decryption_queue.go]: IMO, we should only log errors for something that is "our" fault, never user error.

**Review by KuphJr — APPROVED** [review]

**Pull request merged**